src/neuroconv/tools/nwb_helpers/_dataset_configuration.py

```diff
@@ -0,0 +1,197 @@
+"""Collection of helper functions related to configuration of datasets dependent on backend."""
+from typing import Iterable, Literal, Union
+
+import h5py
+import numpy as np
+import zarr
+from hdmf import Container
+from hdmf.data_utils import DataChunkIterator, DataIO, GenericDataChunkIterator
+from hdmf.utils import get_data_shape
+from hdmf_zarr import NWBZarrIO
+from pynwb import NWBHDF5IO, NWBFile, TimeSeries
+from pynwb.base import DynamicTable
+
+from ._dataset_and_backend_models import (
+    BACKEND_TO_CONFIGURATION,
+    BACKEND_TO_DATASET_CONFIGURATION,
+    DatasetConfiguration,
+    DatasetInfo,
+    HDF5BackendConfiguration,
+    HDF5DatasetConfiguration,
+    ZarrBackendConfiguration,
+    ZarrDatasetConfiguration,
+)
+from ..hdmf import SliceableDataChunkIterator
+
+
+def _get_mode(io: Union[NWBHDF5IO, NWBZarrIO]) -> str:
+    """NWBHDF5IO and NWBZarrIO have different ways of storing the mode they used on a path."""
+    if isinstance(io, NWBHDF5IO):
+        return io.mode
+    elif isinstance(io, NWBZarrIO):
+        return io._ZarrIO__mode
```

Comment:
Can we not use a mangled private variable? If we need this, can we get the `NWBZarrIO` object to expose it properly?

Reply:
Knowing the mode is necessary at https://github.com/catalystneuro/neuroconv/pull/569/files#diff-91679913ea0f31861b6d5e7275fa4a7e03b256d6cd91aa8d82ca505c604d9112R152-R155

Can you open an issue/PR on HDMF Zarr for that? I was unable to figure another way, so this technically temporary if we make the NWBZarrIO have a `.mode` attribute

Reply:
Raised https://github.com/hdmf-dev/hdmf-zarr/issues/137 and will leave a comment in this code that it's a temporary workaround until upstream is fixed and released